traits/tests/check_observe_timing.py

```diff
@@ -0,0 +1,430 @@
+    construct_person = person_construction_template.format(decorator=decorator)
+    construction_time = timeit.timeit(
+        construct_person, base_setup, number=N
+    )
+
+    instantiate_setup = person_instantiation_setup_template.format(
+        decorator=decorator
+    )
+    instantiation_time = timeit.timeit(
+        instantiate_person, instantiate_setup, number=N
+    )
+
+    reassign_setup = person_name_reassignment_setup_template.format(
+        decorator=decorator
+    )
+    reassign_person_name_time = timeit.timeit(
+        reassign_name, reassign_setup, number=N
+    )
+
+    return (construction_time, instantiation_time, reassign_person_name_time)
+
+
+# Code template strings to be timed for scenario 2
+
+construct_parent_setup = base_setup + """
+class Person(HasTraits):
+    name = Str()
+"""
+parent_construction_template = """
+class Parent(HasTraits):
+    child = Instance(Person)
+    {decorator}
+    def some_method(self, event):
+        pass
+"""
+
+parent_instantiation_setup_template = construct_parent_setup \
+    + parent_construction_template
+instantiate_parent = """
+Parent()
+"""
+
+child_reassignment_setup_template = parent_instantiation_setup_template \
+    + "a_parent = Parent(child=Person())"
+reassign_child = """
+a_parent.child = Person()
+"""
+
+reassign_child_name = """
+a_parent.child.name = ''
+"""
+
+
+def scenario2(decorator):
+    """ Time the cases described by scenario 2 using the given method decorator.
+
+    Parameters
+    ----------
+    decorator - Str
+        The string defining the decorator to be used on the method of the
+        HasTraits subclass.  e.g. "@observe('child.name')"
+
+    Returns
+    -------
+    tuple
+        A 4-tuple containing the time to construct the HasTraits subclass, the
+        time to instantiate it, the time to reassign child, and the
+        time to reassign child.name
+    """
+    construct_parent = parent_construction_template.format(
+        decorator=decorator
+    )
+    construction_time = timeit.timeit(
+        construct_parent, construct_parent_setup, number=N
+    )
+
+    instantiate_parent_setup = parent_instantiation_setup_template.format(
+        decorator=decorator
+    )
+    instantiation_time = timeit.timeit(
+        instantiate_parent, instantiate_parent_setup, number=N
+    )
+
+    reassign_child_setup = child_reassignment_setup_template.format(
+        decorator=decorator
+    )
+    reassign_child_time = timeit.timeit(
+        reassign_child, reassign_child_setup, number=N
+    )
+
+    reassign_child_name_time = timeit.timeit(
+        reassign_child_name, reassign_child_setup, number=N
+    )
+
+    return (
+        construction_time,
+        instantiation_time,
+        reassign_child_time,
+        reassign_child_name_time
+    )
+
+
+# Code template strings to be timed for scenario 3
+
+person_with_property_construction_template = """
+class Person(HasTraits):
+    name=Str()
+    a_property = Property({})
```

Comment:
A property without a getter is a rather unrealistic setup. 
I don't think it would change the answer, but to put that wonder to rest, let's add a getter for this.
e.g.
```
def _get_a_property(self):
    return self.name
```